### ISSUE TITLE
INTERNAL ERROR: Black produced code that is not equivalent to the source.  Please report a bug

### ISSUE THREAD

**Issue by ceeb3** [psf/black#1381]
**Describe the bug** A clear and concise description of what the bug is.
I was blackening simple python files under 100 lines each.  I got the above error on one file.  
**To Reproduce** Steps to reproduce the behavior:

The diff is:

```
--- src
+++ dst
@@ -140,11 +140,11 @@
       value=
         Call(
           args=
             Constant(
               value=
-                b'main',  # bytes
+                'main',  # str
             )  # /Constant
             Name(
               ctx=
                 Load(
                 )  # /Load
@@ -158,57 +158,50 @@
                 )  # /Load
               id=
                 'Blueprint',  # str
             )  # /Name
           keywords=
-          kwargs=
-            None,  # NoneType
-          starargs=
-            None,  # NoneType
         )  # /Call
     )  # /Assign
     FunctionDef(
       args=
         arguments(
           args=
           defaults=
+          kw_defaults=
           kwarg=
             None,  # NoneType
-          type_comments=
+          kwonlyargs=
           vararg=
             None,  # NoneType
         )  # /arguments
       body=
         Return(
           value=
             Call(
               args=
                 Constant(
                   value=
-                    b'password_reset_email.html',  # bytes
+                    'password_reset_email.html',  # str
                 )  # /Constant
               func=
                 Name(
                   ctx=
                     Load(
                     )  # /Load
                   id=
                     'render_template',  # str
                 )  # /Name
               keywords=
-              kwargs=
-                None,  # NoneType
-              starargs=
-                None,  # NoneType
             )  # /Call
         )  # /Return
       decorator_list=
         Call(
           args=
             Constant(
               value=
-                b'/',  # bytes
+                '/',  # str
             )  # /Constant
           func=
             Attribute(
               attr=
                 'route',  # str
@@ -223,61 +216,56 @@
                   id=
                     'main',  # str
                 )  # /Name
             )  # /Attribute
           keywords=
-          kwargs=
-            None,  # NoneType
-          starargs=
-            None,  # NoneType
         )  # /Call
       name=
         'index',  # str
+      returns=
+        None,  # NoneType
       type_comment=
         None,  # NoneType
     )  # /FunctionDef
     FunctionDef(
       args=
         arguments(
           args=
           defaults=
+          kw_defaults=
           kwarg=
             None,  # NoneType
-          type_comments=
+          kwonlyargs=
           vararg=
             None,  # NoneType
         )  # /arguments
       body=
         Return(
           value=
             Call(
               args=
                 Constant(
                   value=
-                    b'get_user_id_email.html',  # bytes
+                    'get_user_id_email.html',  # str
                 )  # /Constant
               func=
                 Name(
                   ctx=
                     Load(
                     )  # /Load
                   id=
                     'render_template',  # str
                 )  # /Name
               keywords=
-              kwargs=
-                None,  # NoneType
-              starargs=
-                None,  # NoneType
             )  # /Call
         )  # /Return
       decorator_list=
         Call(
           args=
             Constant(
               value=
-                b'/user_id',  # bytes
+                '/user_id',  # str
             )  # /Constant
           func=
             Attribute(
               attr=
                 'route',  # str
@@ -292,28 +280,27 @@
                   id=
                     'main',  # str
                 )  # /Name
             )  # /Attribute
           keywords=
-          kwargs=
-            None,  # NoneType
-          starargs=
-            None,  # NoneType
         )  # /Call
       name=
         'get_user_id',  # str
+      returns=
+        None,  # NoneType
       type_comment=
         None,  # NoneType
     )  # /FunctionDef
     FunctionDef(
       args=
         arguments(
           args=
           defaults=
+          kw_defaults=
           kwarg=
             None,  # NoneType
-          type_comments=
+          kwonlyargs=
           vararg=
             None,  # NoneType
         )  # /arguments
       body=
         Assign(
@@ -330,11 +317,11 @@
           value=
             Call(
               args=
                 Constant(
                   value=
-                    b'email',  # bytes
+                    'email',  # str
                 )  # /Constant
               func=
                 Attribute(
                   attr=
                     'get',  # str
@@ -357,14 +344,10 @@
                             'request',  # str
                         )  # /Name
                     )  # /Attribute
                 )  # /Attribute
               keywords=
-              kwargs=
-                None,  # NoneType
-              starargs=
-                None,  # NoneType
             )  # /Call
         )  # /Assign
         Assign(
           targets=
             Name(
@@ -379,11 +362,11 @@
           value=
             Call(
               args=
                 Constant(
                   value=
-                    b'HTTP_X_REAL_IP',  # bytes
+                    'HTTP_X_REAL_IP',  # str
                 )  # /Constant
                 Attribute(
                   attr=
                     'remote_addr',  # str
                   ctx=
@@ -421,14 +404,10 @@
                             'request',  # str
                         )  # /Name
                     )  # /Attribute
                 )  # /Attribute
               keywords=
-              kwargs=
-                None,  # NoneType
-              starargs=
-                None,  # NoneType
             )  # /Call
         )  # /Assign
         Assign(
           targets=
             Name(
@@ -458,17 +437,13 @@
                       id=
                         'datetime',  # str
                     )  # /Name
                 )  # /Attribute
               keywords=
-              kwargs=
-                None,  # NoneType
-              starargs=
-                None,  # NoneType
             )  # /Call
         )  # /Assign
-        TryExcept(
+        Try(
           body=
             Assign(
               targets=
                 Name(
                   ctx=
@@ -527,23 +502,16 @@
                                     )  # /Load
                                   id=
                                     'email',  # str
                                 )  # /Name
                             )  # /keyword
-                          kwargs=
-                            None,  # NoneType
-                          starargs=
-                            None,  # NoneType
                         )  # /Call
                     )  # /Attribute
                   keywords=
-                  kwargs=
-                    None,  # NoneType
-                  starargs=
-                    None,  # NoneType
                 )  # /Call
             )  # /Assign
+          finalbody=
           handlers=
             ExceptHandler(
               body=
                 Assign(
                   targets=
@@ -600,14 +568,10 @@
                                     )  # /Load
                                   id=
                                     'str',  # str
                                 )  # /Name
                               keywords=
-                              kwargs=
-                                None,  # NoneType
-                              starargs=
-                                None,  # NoneType
                             )  # /Call
                         )  # /keyword
                         keyword(
                           arg=
                             'eventDate',  # str
@@ -628,20 +592,12 @@
                                     )  # /Load
                                   id=
                                     'str',  # str
                                 )  # /Name
                               keywords=
-                              kwargs=
-                                None,  # NoneType
-                              starargs=
-                                None,  # NoneType
                             )  # /Call
                         )  # /keyword
-                      kwargs=
-                        None,  # NoneType
-                      starargs=
-                        None,  # NoneType
                     )  # /Call
                 )  # /Assign
                 Expr(
                   value=
                     Call(
@@ -676,14 +632,10 @@
                                     'db',  # str
                                 )  # /Name
                             )  # /Attribute
                         )  # /Attribute
                       keywords=
-                      kwargs=
-                        None,  # NoneType
-                      starargs=
-                        None,  # NoneType
                     )  # /Call
                 )  # /Expr
                 Expr(
                   value=
                     Call(
@@ -711,85 +663,67 @@
                                     'db',  # str
                                 )  # /Name
                             )  # /Attribute
                         )  # /Attribute
                       keywords=
-                      kwargs=
-                        None,  # NoneType
-                      starargs=
-                        None,  # NoneType
                     )  # /Call
                 )  # /Expr
                 Expr(
                   value=
                     Call(
                       args=
                         Constant(
                           value=
-                            b'Please check your email for your user name.',  # bytes
+                            'Please check your email for your user name.',  # str
                         )  # /Constant
                         Constant(
                           value=
-                            b'form-success',  # bytes
+                            'form-success',  # str
                         )  # /Constant
                       func=
                         Name(
                           ctx=
                             Load(
                             )  # /Load
                           id=
                             'flash',  # str
                         )  # /Name
                       keywords=
-                      kwargs=
-                        None,  # NoneType
-                      starargs=
-                        None,  # NoneType
                     )  # /Call
                 )  # /Expr
                 Return(
                   value=
                     Call(
                       args=
                         Constant(
                           value=
-                            b'password_reset_email.html',  # bytes
+                            'password_reset_email.html',  # str
                         )  # /Constant
                       func=
                         Name(
                           ctx=
                             Load(
                             )  # /Load
                           id=
                             'render_template',  # str
                         )  # /Name
                       keywords=
-                      kwargs=
-                        None,  # NoneType
-                      starargs=
-                        None,  # NoneType
                     )  # /Call
                 )  # /Return
               name=
-                Name(
-                  ctx=
-                    Store(
-                    )  # /Store
-                  id=
-                    'e',  # str
-                )  # /Name
+                'e',  # str
               type=
                 Name(
                   ctx=
                     Load(
                     )  # /Load
                   id=
                     'Exception',  # str
                 )  # /Name
             )  # /ExceptHandler
           orelse=
-        )  # /TryExcept
+        )  # /Try
         If(
           body=
             Expr(
               value=
                 Call(
@@ -808,14 +742,10 @@
                         )  # /Load
                       id=
                         'send_password_reset_email',  # str
                     )  # /Name
                   keywords=
-                  kwargs=
-                    None,  # NoneType
-                  starargs=
-                    None,  # NoneType
                 )  # /Call
             )  # /Expr
             Assign(
               targets=
                 Name(
@@ -899,23 +829,19 @@
                                 )  # /Load
                               id=
                                 'str',  # str
                             )  # /Name
                           keywords=
-                          kwargs=
-                            None,  # NoneType
-                          starargs=
-                            None,  # NoneType
                         )  # /Call
                     )  # /keyword
                     keyword(
                       arg=
                         'type',  # str
                       value=
                         Constant(
                           value=
-                            b'get_user_id',  # bytes
+                            'get_user_id',  # str
                         )  # /Constant
                     )  # /keyword
                     keyword(
                       arg=
                         'eventDate',  # str
@@ -936,20 +862,12 @@
                                 )  # /Load
                               id=
                                 'str',  # str
                             )  # /Name
                           keywords=
-                          kwargs=
-                            None,  # NoneType
-                          starargs=
-                            None,  # NoneType
                         )  # /Call
                     )  # /keyword
-                  kwargs=
-                    None,  # NoneType
-                  starargs=
-                    None,  # NoneType
                 )  # /Call
             )  # /Assign
             Expr(
               value=
                 Call(
@@ -984,14 +902,10 @@
                                 'db',  # str
                             )  # /Name
                         )  # /Attribute
                     )  # /Attribute
                   keywords=
-                  kwargs=
-                    None,  # NoneType
-                  starargs=
-                    None,  # NoneType
                 )  # /Call
             )  # /Expr
             Expr(
               value=
                 Call(
@@ -1019,14 +933,10 @@
                                 'db',  # str
                             )  # /Name
                         )  # /Attribute
                     )  # /Attribute
                   keywords=
-                  kwargs=
-                    None,  # NoneType
-                  starargs=
-                    None,  # NoneType
                 )  # /Call
             )  # /Expr
           orelse=
           test=
             Name(
@@ -1040,11 +950,11 @@
       decorator_list=
         Call(
           args=
             Constant(
               value=
-                b'/user_id',  # bytes
+                '/user_id',  # str
             )  # /Constant
           func=
             Attribute(
               attr=
                 'route',  # str
@@ -1070,21 +980,19 @@
                     Load(
                     )  # /Load
                   elts=
                     Constant(
                       value=
-                        b'POST',  # bytes
+                        'POST',  # str
                     )  # /Constant
                 )  # /List
             )  # /keyword
-          kwargs=
-            None,  # NoneType
-          starargs=
-            None,  # NoneType
         )  # /Call
       name=
         'get_user_id_post',  # str
+      returns=
+        None,  # NoneType
       type_comment=
         None,  # NoneType
     )  # /FunctionDef
   type_ignores=
 )  # /Module
```

- Version: black, version 19.10b0
- OS and Python version: MacOS 10.14.6 (18G103)

**Comment by ichard26** [comment]
@ceeb3 It would helpful for debugging if you could include the original problematic source code and the steps that lead to the error like what options you ran _Black_ with. The bug looks interesting though, but for me it's a bit hard to parse the above diff output, sorry.

**Comment by tommilligan** [comment]
I believe this is a dupe of #1287, closed by #1315
